### PR TITLE
fix: do not add address if it already exists

### LINKED ISSUE #1504: `address add` may overwrite an existing alias
Using `address add --alias <alias> --address <address>` may re-assign address to an alias different from the alias of the respective key.

### PR DESCRIPTION
**Pull request by karbyshev**
Closes #1504

### CODE REVIEW

File: shared/src/ledger/wallet/store.rs [hunk 1]
```diff
@@ -504,6 +504,15 @@ impl Store {
         address: Address,
         force: bool,
     ) -> Option<Alias> {
+        // abort if the address already exists in the wallet
+        if self.addresses.contains_right(&address) {
```
Comment (tzemanovic) [inline shared/src/ledger/wallet/store.rs:508]: I think this should respect the `force` flag

File: shared/src/ledger/wallet/store.rs [hunk 2]
```diff
@@ -504,6 +504,15 @@ impl Store {
...
+            println!(
+                "Address {} already exists in the wallet.",
```
Comment (tzemanovic) [inline shared/src/ledger/wallet/store.rs:510]: let's also print the alias if it exists

File: shared/src/ledger/wallet/store.rs (hunk 1 above)
Comment (karbyshev) [reply shared/src/ledger/wallet/store.rs]: Thanks! Fixed that.

File: shared/src/ledger/wallet/store.rs (hunk 2 above)
Comment (karbyshev) [reply shared/src/ledger/wallet/store.rs]: Thanks! Added.

**Review by tzemanovic — APPROVED** [review]

**Pull request merged**